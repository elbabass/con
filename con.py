#!/usr/bin/env python
# coding: utf-8

import subprocess
import sys

TRADUCTIONS = {
    u'tracte': u'pull',
    u"pulse": u"push",
    u"forme": u"clone",
    u"bine": u"merge",
    u"fonde": u"rebase",
    u"pare": u"diff",
    u"signe": u"log",
    u"state": u"status",
    u"comitant": u"commit"
}


def traduit_commande(commande):
    if commande in TRADUCTIONS.keys():
        return TRADUCTIONS[commande]
    else:
        return commande


def remplace_commande_principale(commande):
    commande[0] = 'git'
    return commande


def traduit_ligne_commande(ligne_commande):
    return [traduit_commande(arg) for arg in ligne_commande]


def main():
    ligne_commande = remplace_commande_principale(sys.argv)
    command_line_list = traduit_ligne_commande(ligne_commande)
    subprocess.call(command_line_list)


if __name__ == '__main__':  # pragma: no cover
    main()
