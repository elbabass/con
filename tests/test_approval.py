from approvaltests.approvals import verify
from con import (
    traduit_commande, remplace_commande_principale, traduit_ligne_commande,
)



def test_examples():
    input = ["tracte",
            "pulse",
            "forme",
            "bine",
            "fonde",
            "pare",
            "signe",
            "state",
            "comitant"]
    verify("\n".join([genere_traduction(nom) for nom in input]))

def genere_traduction(nom):
  actual = traduit_commande(nom)
  return f"* ``git {actual}`` â†’ ``con {nom}``"

