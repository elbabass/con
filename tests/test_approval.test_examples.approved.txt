* ``git pull`` → ``con tracte``
* ``git push`` → ``con pulse``
* ``git clone`` → ``con forme``
* ``git merge`` → ``con bine``
* ``git rebase`` → ``con fonde``
* ``git diff`` → ``con pare``
* ``git log`` → ``con signe``
* ``git status`` → ``con state``
* ``git commit`` → ``con comitant``
